src/components/MovieDetail/MovieDetail.js

```diff
@@ -1,41 +1,72 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './MovieDetail.css';
 import { BsPlayFill } from 'react-icons/bs';
+import { getSelectedMovie, handleErrors } from '../../utilities';
 
-const MovieDetail = ({ details }) => {
-  const percentage = details.average_rating * 10;
-  const ratingStyle = {
-    background: `linear-gradient(to right, #ffd000 ${percentage}%, #cfe2f7d8 ${percentage}%)`
-  };
+class MovieDetail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...props };
+  }
 
-  return (
-    <div className="movie-detail">
-      <section className="movie-card__details">
-        <img src={details.poster_path} className="movie-card__details__poster" />
-        <h2 className="movie-card__details__title">{details.title}</h2>
-        <h3 className="movie-card__details__date">{details.release_date.split('-')[0]}</h3>
-        <div
-          className="movie-card__details__rating"
-          style={ratingStyle}
-        >
-          <span className="fa fa-star"></span>
-          <span className="fa fa-star"></span>
-          <span className="fa fa-star"></span>
-          <span className="fa fa-star"></span>
-          <span className="fa fa-star"></span>
+  componentDidMount = () => {
+    const { id } = this.state;
+
+    getSelectedMovie(id)
+      .then((response) => {
+        this.props.updateFetchStatus('fetchStatus', response.status);
+        return handleErrors(response);
+      })
+      .then((movieData) => this.setState({ ...movieData.movie }))
+      .catch(() => this.props.updateFetchStatus('fetchError', true));
+  }
+
+  render() {
+    const {
+      average_rating,
+      title,
+      release_date,
+      tagline,
+      runtime,
+      genres,
+      overview,
+      backdrop_path,
+      poster_path
+    } = this.state;
+    const percentage = average_rating * 10;
+    const ratingStyle = {
+      background: `linear-gradient(to right, #ffd000 ${percentage}%, #cfe2f7d8 ${percentage}%)`
+    };
+
+    return (
+      <main className="movie-detail">
+        <section className="movie-card__details">
+          <img src={poster_path} className="movie-card__details__poster" />
+          <h2 className="movie-card__details__title">{title}</h2>
+          <h3 className="movie-card__details__date">{release_date?.split('-')[0]}</h3>
+          <div
+            className="movie-card__details__rating"
+            style={ratingStyle}
+          >
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
+            <span className="fa fa-star" />
```

Comment:
I'm looking into a solution to this wetness I think there may be a few options

Reply:
Nice!